src/custom-components/ReorderableList.tsx

```diff
@@ -0,0 +1,127 @@
+import { Fragment, JSXElementConstructor, ReactElement, useEffect, useState } from "react";
+import { Field, FieldProps, Focusable, GamepadButton } from "../deck-components";
+
+export type ReorderableEntry<T> = {
+  label: string,
+  data?:T,
+  position:number
+}
+
+type ListProps<T> = {
+  entries: ReorderableEntry<T>[],
+  onSave: (entries: ReorderableEntry<T>[]) => void,
+  interactables?: JSXElementConstructor<{entry:ReorderableEntry<T>}>,
+  fieldProps?: FieldProps
+}
+
+/**
+ * A component for creating reorderable lists.
+ * 
+ * Implementation example can be found {@link https://github.com/Tormak9970/Component-Testing-Plugin/blob/main/src/testing-window/ReorderableListTest.tsx here}.
+ */
+export function ReorderableList<T>(props: ListProps<T>) {
+  const [entryList, setEntryList] = useState<ReorderableEntry<T>[]>(props.entries.sort((a:ReorderableEntry<T>, b:ReorderableEntry<T>) => a.position - b.position));
+  const [reorderEnabled, setReorderEnabled] = useState<boolean>(false);
+
+  useEffect(() => {
+    setEntryList(props.entries.sort((a: ReorderableEntry<T>, b: ReorderableEntry<T>) => a.position - b.position));
+  }, [props.entries]);
+
+  function toggleReorderEnabled(): void {
+    let newReorderValue = !reorderEnabled;
+    setReorderEnabled(newReorderValue);
+
+    if (!newReorderValue){
+      props.onSave(entryList);
+    }
+  }
+
+  return (
+    <Fragment>
+      <style>{`
```

Comment:
inline style probably better here, we don't want dfl to mess with global CSS too much